Docs: update annotations docs to include typeahead support for tags

axelavargas opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:

Updates documentation for annotations and new HTTP endpoint, now we support typeahead suggestions, images are in these PR https://github.com/grafana/website/pull/5196

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

For #36107

**Special notes for your reviewer**:

axelavargas committed:
Docs: update annotations docs to include typeahead support for tags

axelavargas committed:
Docs: Update Http API annotations including new endpoint to query by tags

marefr commented on docs/sources/http_api/annotations.md:
````diff
@@ -251,3 +251,37 @@ Content-Type: application/json
+
+`GET /api/annotations/tags`
+
+Find all the event tags created in the annotations
+
+**Example Request**:
+
+
+```http
+GET /api/annotations/tags HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Basic YWRtaW46YWRtaW4=
+```
+
+**Example Response**:
+
+```http
+HTTP/1.1 200
+Content-Type: application/json
+
+{
+    "result": {
+        "tags": [
+            {
+                "tag": "outage",
+                "count": 1
+            }
+        ]
+    }
+}
+```
````
A bit unsure if the tag-query is optional or not. Please verify.

```suggestion
Find all the event tags created in the annotations

**Example Request**:

```http
GET /api/annotations/tags?tag=out HTTP/1.1
Accept: application/json
Content-Type: application/json
Authorization: Basic YWRtaW46YWRtaW4=
```

Query Parameters:
- `tag`: string. Optional. Used to filter tags.
- `limit`: number. Optional - default is 100. Max limit for results returned.

**Example Response**:

```http
HTTP/1.1 200
Content-Type: application/json

{
    "result": {
        "tags": [
            {
                "tag": "outage",
                "count": 1
            }
        ]
    }
}
```
```

axelavargas replied:
@marefr Thank you for taking a look 🙌🏾, indeed the tag query is optional (I tested it), I will apply the suggestions

marefr reviewed (changes requested):
Suggested some minor changes

axelavargas committed:
Update docs/sources/http_api/annotations.md

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

axelavargas committed:
Docs: update http-api docs for annotations, added missing query parameters for finding tags

marefr reviewed (approved):
From technical standpoint LGTM, but would ask for technical writers input as well

achatterjee-grafana commented on docs/sources/http_api/annotations.md:
```diff
@@ -251,3 +251,41 @@ Content-Type: application/json
+
+Query Parameters:
+
+- `tag`: string. Optional. Used to filter tags.
```
```suggestion
- `tag`: Optional. A string that you can use to filter tags.
```

achatterjee-grafana commented on docs/sources/http_api/annotations.md:
```diff
@@ -251,3 +251,41 @@ Content-Type: application/json
+Query Parameters:
+
+- `tag`: string. Optional. Used to filter tags.
+- `limit`: number. Optional - default is 100. Max limit for results returned.
```
```suggestion
- `limit`: Optional. A number, where the default is 100. Max limit for results returned.
```

achatterjee-grafana commented on docs/sources/dashboards/annotations.md:
```diff
@@ -41,12 +41,16 @@ can still show them if you add a new **Annotation Query** and filter by tags. Bu
-You can create new annotation queries that fetch annotations from the native annotation store via the `-- Grafana --` data source and by setting *Filter by* to `Tags`. Specify at least
-one tag. For example create an annotation query name `outages` and specify a tag named `outage`. This query will show all annotations you create (from any dashboard or via API) that have the `outage` tag. By default, if you add multiple tags in the annotation query, Grafana will only show annotations that have all the tags you supplied. You can invert the behavior by enabling `Match any` which me ...
+
+You can create new annotation queries that fetch annotations from the native annotation store via the `-- Grafana --` data source and by setting *Filter by* to `Tags`. In Grafana v8.1+ it's possible to get typeahead support of existing tags, you should specify at least one tag.
+
+For example create an annotation query name `outages` and specify a tag named `outage`. This query will show all annotations you create (from any dashboard or via API) that have the `outage` tag. By default, if you add multiple tags in the annotation query, Grafana will only show annotations that have all the tags you supplied. You can invert the behavior by enabling `Match any` which means that  ...
```
```suggestion
You can create new queries to fetch annotations from the native annotation store via the `-- Grafana --` data source by setting *Filter by* to `Tags`. 

Grafana v8.1 and later versions also support typeahead of existing tags provided you have defined at least one tag.

For example, create an annotation query name `outages` and specify a tag `outage`. This query will show all annotations (from any dashboard or via API) with the `outage` tag. If multiple tags are defined in an annotation query, then Grafana will only show annotations matching all the tags. To modify the behavior, enable `Match any`, and Grafana will show annotations that contain any one of the tags you provided.
```

achatterjee-grafana reviewed (commented):
Added doc review.

axelavargas committed:
Apply suggestions from code review

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

axelavargas committed:
Docs: Update image references for annotations

axelavargas commented on docs/sources/http_api/annotations.md:
````diff
@@ -251,3 +251,41 @@ Content-Type: application/json
+Authorization: Basic YWRtaW46YWRtaW4=
+```
+
+Query Parameters:
````
🤔 I don't think so, the Query parameters section is only describing a list of parameters that the API accepts.

achatterjee-grafana replied:
Fair enough.

axelavargas commented:
> Added doc review.

Thank you for the review @achatterjee-grafana, I have updated the docs with your suggestions and also updated the reference of the images :).

achatterjee-grafana reviewed (approved):
LGTM

axelavargas committed:
Merge branch 'main' into axelav/docs-annotations
